plugins/woocommerce-admin/client/products/sections/images-section.tsx

```diff
@@ -0,0 +1,205 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import {
+	Link,
+	useFormContext,
+	MediaUploader,
+	ImageGallery,
+	ImageGalleryItem,
+} from '@woocommerce/components';
+import { Card, CardBody, DropZone } from '@wordpress/components';
+import { recordEvent } from '@woocommerce/tracks';
+import { useState } from '@wordpress/element';
+import { Product } from '@woocommerce/data';
+import classnames from 'classnames';
+import { Icon, trash } from '@wordpress/icons';
+import { MediaItem } from '@wordpress/media-utils';
+
+/**
+ * Internal dependencies
+ */
+import { ProductSectionLayout } from '../layout/product-section-layout';
+import DragAndDrop from '../images/drag-and-drop.svg';
+import './images-section.scss';
+
+type Image = MediaItem & {
+	src: string;
+};
+
+export const ImagesSection: React.FC = () => {
+	const { getInputProps, setValue } = useFormContext< Product >();
+	const images = ( getInputProps( 'images' ).value as Image[] ) || [];
+	const [ isRemoveZoneVisible, setIsRemoveZoneVisible ] =
+		useState< boolean >( false );
+	const [ isRemove, setIsRemove ] = useState< boolean >( false );
+	const [ draggedImageId, setDraggedImageId ] = useState< number | null >(
+		0
+	);
+
+	const toggleRemoveZone = () => {
+		setIsRemoveZoneVisible( ! isRemoveZoneVisible );
+	};
+
+	const orderImages = ( newOrder: JSX.Element[] ) => {
+		const orderedImages = newOrder.map( ( image ) => {
+			return images.find(
+				( file ) =>
+					file.id === parseInt( image?.props?.id, 10 )
+			);
+		} );
+		setValue( 'images', orderedImages );
+	};
+
+	return (
+		<ProductSectionLayout
+			title={ __( 'Images', 'woocommerce' ) }
+			description={
+				<>
+					<span>
+						{ __(
+							'For best results, use JPEG files that are 1000 by 1000 pixels or larger.',
+							'woocommerce'
+						) }
+					</span>
+					<Link
+						className="woocommerce-form-section__header-link"
+						href="https://woocommerce.com/posts/fast-high-quality-product-photos/"
+						target="_blank"
+						type="external"
+						onClick={ () => {
+							recordEvent( 'prepare_images_help' );
+						} }
+					>
+						{ __( 'How should I prepare images?', 'woocommerce' ) }
+					</Link>
+				</>
+			}
+		>
+			<Card
+				className={ classnames( 'woocommerce-product-form__images', {
+					'has-images': images.length > 0,
+				} ) }
+			>
+				<CardBody>
+					<ImageGallery
+						onDragStart={ ( event ) => {
+							const { id: imageId, dataset } = event.target as HTMLElement;
+							if ( imageId ) {
+								setDraggedImageId( parseInt( imageId, 10 ) );
+							} else {
```

Comment:
Just curious - what scenario do we not have `imageId` and end up inside this `else`?

Reply:
That happens when the image to move is grabbed by the icon in the toolbar. At that point, we don't have the image id, and this is why we use the image index (in the `images` list), to get the image id.

![Screen Shot 2022-10-12 at 11 49 17](https://user-images.githubusercontent.com/1314156/195375417-f6752e18-0a91-4804-9624-0ddfe9e320e7.png)